client/packages/system/src/Patient/DetailView/DetailView.tsx

```diff
@@ -1,15 +1,58 @@
 import React, { FC } from 'react';
 import {
   DetailViewSkeleton,
+  SaveAction,
   useJsonForms,
   useUrlQuery,
 } from '@openmsupply-client/common';
+import { usePatient } from '../api';
+import { DocumentFragment } from 'packages/common/src/ui/forms/JsonForms/api';
+
+const useUpsertPatient = (): SaveAction => {
```

Comment:
@CarlosNZ  This hook is to configure how documents are saved in the JSON form. For this I need to propagate the isLoading and isError flags down to useJsonForms. This feels a bit cumbersome, any suggestions?

Reply:
Ideally, you would just have the "mutate" method return a status.

So in the JSONForms "saveData" method, you'd just do something like:

```
const saveData = async () => {
 ... safety checks

 setSaving(true)

const result = await mutate(data, etc...)

setSaving(false)

if (result.error) {
 setError(result.error)
 return
}

setIsDirty(false)
...sucess code (Snack etc.)
OnSaveCallback()
}
```

So rather than passing in `saving`/`loading` and `isError`, you'd just pass the mutation and an "onSave" callback that runs on successful save (and maybe an "onError" callback too).

Reply:
Yeah, it looks like the `await mutate(...)` pattern has been used in plenty of places elsewhere, so you should be able to copy that.

Reply:
great thanks! missed that its returns a Promise